src/devices/Arduino/ArduinoAnalogController.cs

```diff
@@ -0,0 +1,101 @@
+        private readonly Dictionary<int, ValueChangedEventHandler> _callbacks;
+
+        public ArduinoAnalogController(ArduinoBoard board,
+            ReadOnlyCollection<SupportedPinConfiguration> supportedPinConfigurations, PinNumberingScheme scheme)
+            : base(scheme)
+        {
+            _board = board ?? throw new ArgumentNullException(nameof(board));
+            _supportedPinConfigurations = supportedPinConfigurations ?? throw new ArgumentNullException(nameof(supportedPinConfigurations));
+            _callbacks = new Dictionary<int, ValueChangedEventHandler>();
+            PinCount = _supportedPinConfigurations.Count;
+
+            // Note: While the Arduino does have an external analog input reference pin, Firmata doesn't allow configuring it.
+            VoltageReference = 5.0;
+        }
+
+        public override int PinCount
+        {
+            get;
+        }
+
+        public override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber)
+        {
+            int numberAnalogPinsFound = 0;
+            for (int i = 0; i < _supportedPinConfigurations.Count; i++)
+            {
+                if (_supportedPinConfigurations[i].PinModes.Contains(SupportedMode.ANALOG_INPUT))
+                {
+                    numberAnalogPinsFound++;
+                    if (pinNumber == i)
+                    {
+                        return numberAnalogPinsFound - 1;
+                    }
+                }
+            }
+
+            throw new InvalidOperationException($"Pin {pinNumber} is not a valid analog input pin.");
+        }
+
+        public override int ConvertLogicalNumberingSchemeToPinNumber(int logicalPinNumber)
+        {
+            int numberAnalogPinsFound = 0;
+            for (int i = 0; i < _supportedPinConfigurations.Count; i++)
+            {
+                if (_supportedPinConfigurations[i].PinModes.Contains(SupportedMode.ANALOG_INPUT))
+                {
+                    numberAnalogPinsFound++;
+                    if (logicalPinNumber == numberAnalogPinsFound - 1)
+                    {
+                        return i;
+                    }
+                }
+            }
+
+            throw new InvalidOperationException($"Pin A{logicalPinNumber} is not existing");
+        }
+
+        public override bool SupportsAnalogInput(int pinNumber)
+        {
+            return _supportedPinConfigurations[pinNumber].PinModes.Contains(SupportedMode.ANALOG_INPUT);
+        }
+
+        protected override AnalogInputPin OpenPinCore(int pinNumber)
+        {
+            // This method is called with the logical pin numbering (input pin A0 is 0, A1 is 1, etc)
+            // but the SetPinMode method operates on the global numbers
+            int fullNumber = ConvertLogicalNumberingSchemeToPinNumber(pinNumber);
+            _board.Firmata.SetPinMode(fullNumber, SupportedMode.ANALOG_INPUT);
```

Comment:
>_board [](start = 12, length = 6)

Keep in mind that it is possible that _board is already disposed, in which case a bunch of things in here might not work appropriately. That is the same case for the rest of the devices (ArduinoGpioControllerDriver, ArduinoI2cDevice, ArduinoSpiDevice). I think that ArduinoBoard should have a internal bool disposed, and all of these devices that were created using it should throw whenever that has been disposed already.

Reply:
Just tested again: That works already. The `_board.Firmata` property throws an ObjectDisposedException when the board is disposed.

Reply:
I thought you were setting Firmata to null, in any case I think it would be better to not rely on Firmata implementation as that may change, and instead make sure that _board also throws ObjectDisposedException when trying to access it or one of its fields.

Reply:
`Firmata` is a property with a backing field. Only the backing field is set to null.